src/back/spv/writer.rs

```diff
@@ -1610,24 +1635,66 @@ impl Writer {
         }
     }
 
+    fn create_block(&mut self) -> Block {
+        let mut block = Block::default();
+        let label_id = self.generate_id();
+        let label_instruction = super::instructions::instruction_label(label_id);
+        block.label_id = Some(label_id);
+        block.label = Some(label_instruction);
+        block
+    }
+
     fn write_block(
         &mut self,
-        statements: &[crate::Statement],
+        statements: &crate::Block,
         ir_module: &crate::Module,
         ir_function: &crate::Function,
         function: &mut Function,
-    ) -> Result<spirv::Word, Error> {
-        let mut block = Block::default();
-        let id = self.generate_id();
-        block.label = Some(super::instructions::instruction_label(id));
+        branch_id: Option<spirv::Word>,
```

Comment:
should this be called `merge_id`?

Reply:
so if it's None, means we are in the root block, correct? I wonder if we should make it non-optional then. We'd just make `block.termination = Return` for the root block (before calling `write_block`), essentially blocking the effect of the branch ID.

Reply:
Yes, `merge_id` would be a better name.

To follow up on your second comment: No I don't think so. As Blocks always require a termination (following the spec), essentially it always has to have either a `merge_id` or a termination instruction from other terminations (only case is `Return` that I can think of right now, that is a statement).